scene/debugger/scene_debugger.cpp

```diff
@@ -127,6 +133,43 @@ Error SceneDebugger::parse_message(void *p_user, const String &p_msg, const Arra
 			scene_tree->get_root()->set_camera_3d_override_orthogonal(size_or_fov, depth_near, depth_far);
 		}
 		scene_tree->get_root()->set_camera_3d_override_transform(transform);
+
+	} else if (p_msg == "add_preview_nodes_to_current_3d_scene") {
+		ERR_FAIL_COND_V(p_args.size() < 6, ERR_INVALID_DATA);
+		const bool should_add_world_environment = p_args[0];
+		const bool should_add_directional_light_3d = p_args[1];
+		const bool should_add_camera_3d = p_args[2];
+
+		if (should_add_world_environment) {
+			WARN_PRINT_ED("Current scene was run without a WorldEnvironment node, adding preview WorldEnvironment to the scene. You can disable this by changing `editor/run/automatically_add_preview_nodes_when_running_single_3d_scene` in Project Settings.");
+
+			WorldEnvironment *world_environment = Object::cast_to<WorldEnvironment>(VariantUtilityFunctions::bytes_to_var_with_objects(p_args[3]));
+			ERR_FAIL_COND_V(world_environment == nullptr, ERR_INVALID_DATA);
+
+			scene_tree->get_root()->add_child(world_environment, false, Node::INTERNAL_MODE_BACK);
+			world_environment->set_name("WorldEnvironmentPreview");
+		}
+
+		if (should_add_directional_light_3d) {
+			WARN_PRINT_ED("Current scene was run without a DirectionalLight3D node, adding preview DirectionalLight3D to the scene. You can disable this by changing `editor/run/automatically_add_preview_nodes_when_running_single_3d_scene` in Project Settings.");
```

Comment:
```suggestion
			WARN_PRINT_ED("Current scene was run without a DirectionalLight3D node; adding preview DirectionalLight3D to the scene. This will not carry over to the exported project.\nYou can disable this by changing `editor/run/automatically_add_preview_nodes_when_running_single_3d_scene` in Project Settings.");
```